```
--- FAIL: TestExport (0.00s)
    otlp_test.go:63: 
        	Error Trace:	otlp_test.go:63
        	Error:      	Not equal: 
        	            	expected: pdata.Metrics{orig:(*v1.ExportMetricsServiceRequest)(0xc000486060)}
        	            	actual  : pdata.Metrics{orig:(*v1.ExportMetricsServiceRequest)(0xc000116060)}
        	            	
        	            	Diff:
        	Test:       	TestExport
FAIL
coverage: 13.3
FAIL	go.opentelemetry.io/collector/receiver/otlpreceiver/internal/metrics	0.015s
```

--- a/model/internal/otlp_wrapper.go
+++ b/model/internal/otlp_wrapper.go
@@ -18,6 +18,7 @@ import (
 	otlpcollectorlog "go.opentelemetry.io/collector/model/internal/data/protogen/collector/logs/v1"
 	otlpcollectormetrics "go.opentelemetry.io/collector/model/internal/data/protogen/collector/metrics/v1"
 	otlpcollectortrace "go.opentelemetry.io/collector/model/internal/data/protogen/collector/trace/v1"
+	otlpcommon "go.opentelemetry.io/collector/model/internal/data/protogen/common/v1"
 	otlpmetrics "go.opentelemetry.io/collector/model/internal/data/protogen/metrics/v1"
 	otlptrace "go.opentelemetry.io/collector/model/internal/data/protogen/trace/v1"
 )
@@ -44,6 +45,7 @@ func MetricsFromOtlp(req *otlpcollectormetrics.ExportMetricsServiceRequest) Metr
 // - Convert IntHistogram to Histogram. See https://github.com/open-telemetry/opentelemetry-proto/blob/f3b0ee0861d304f8f3126686ba9b01c106069cb0/opentelemetry/proto/metrics/v1/metrics.proto#L170
 // - Convert IntGauge to Gauge. See https://github.com/open-telemetry/opentelemetry-proto/blob/f3b0ee0861d304f8f3126686ba9b01c106069cb0/opentelemetry/proto/metrics/v1/metrics.proto#L156
 // - Convert IntSum to Sum. See https://github.com/open-telemetry/opentelemetry-proto/blob/f3b0ee0861d304f8f3126686ba9b01c106069cb0/opentelemetry/proto/metrics/v1/metrics.proto#L156
+// - Converts Labels to Attributes. See https://github.com/open-telemetry/opentelemetry-proto/blob/8672494217bfc858e2a82a4e8c623d4a5530473a/opentelemetry/proto/metrics/v1/metrics.proto#L385
 func MetricsCompatibilityChanges(req *otlpcollectormetrics.ExportMetricsServiceRequest) {
 	for _, rsm := range req.ResourceMetrics {
 		for _, ilm := range rsm.InstrumentationLibraryMetrics {
@@ -55,6 +57,14 @@ func MetricsCompatibilityChanges(req *otlpcollectormetrics.ExportMetricsServiceR
 					metric.Data = intGaugeToGauge(m)
 				case *otlpmetrics.Metric_IntSum:
 					metric.Data = intSumToSum(m)
+				case *otlpmetrics.Metric_Sum:
+					numberDataPointsLabelsToAttributes(m.Sum.DataPoints)
+				case *otlpmetrics.Metric_Gauge:
+					numberDataPointsLabelsToAttributes(m.Gauge.DataPoints)
+				case *otlpmetrics.Metric_Summary:
+					summaryDataPointsLabelsToAttributes(m.Summary.DataPoints)
+				case *otlpmetrics.Metric_Histogram:
+					histogramDataPointsLabelsToAttributes(m.Histogram.DataPoints)
 				default:
 				}
 			}
@@ -121,6 +131,21 @@ func LogsFromOtlp(req *otlpcollectorlog.ExportLogsServiceRequest) LogsWrapper {
 	return LogsWrapper{req: req}
 }
 
+func labelsToAttributes(labels []otlpcommon.StringKeyValue) []otlpcommon.KeyValue { //nolint:staticcheck // SA1019 ignore this!
+	attrs := make([]otlpcommon.KeyValue, len(labels))
+	for i, v := range labels {
+		attrs[i] = otlpcommon.KeyValue{
+			Key: v.Key,
+			Value: otlpcommon.AnyValue{
+				Value: &otlpcommon.AnyValue_StringValue{
+					StringValue: v.Value,
+				},
+			},
+		}
+	}
+	return attrs
+}
+
 func intHistogramToHistogram(src *otlpmetrics.Metric_IntHistogram) *otlpmetrics.Metric_Histogram {
 	datapoints := []*otlpmetrics.HistogramDataPoint{}
 	for _, datapoint := range src.IntHistogram.DataPoints {
@@ -133,6 +158,7 @@ func intHistogramToHistogram(src *otlpmetrics.Metric_IntHistogram) *otlpmetrics.
 			BucketCounts:      datapoint.BucketCounts,
 			ExplicitBounds:    datapoint.ExplicitBounds,
 			Exemplars:         intExemplarToExemplar(datapoint.Exemplars),
+			Attributes:        labelsToAttributes(datapoint.Labels),
 		})
 	}
 	return &otlpmetrics.Metric_Histogram{
@@ -152,6 +178,7 @@ func intGaugeToGauge(src *otlpmetrics.Metric_IntGauge) *otlpmetrics.Metric_Gauge
 			StartTimeUnixNano: datapoint.StartTimeUnixNano,
 			Exemplars:         intExemplarToExemplar(datapoint.Exemplars),
 			Value:             &otlpmetrics.NumberDataPoint_AsInt{AsInt: datapoint.Value},
+			Attributes:        labelsToAttributes(datapoint.Labels),
 		}
 	}
 	return &otlpmetrics.Metric_Gauge{
@@ -170,6 +197,7 @@ func intSumToSum(src *otlpmetrics.Metric_IntSum) *otlpmetrics.Metric_Sum {
 			StartTimeUnixNano: datapoint.StartTimeUnixNano,
 			Exemplars:         intExemplarToExemplar(datapoint.Exemplars),
 			Value:             &otlpmetrics.NumberDataPoint_AsInt{AsInt: datapoint.Value},
+			Attributes:        labelsToAttributes(datapoint.Labels),
 		}
 	}
 	return &otlpmetrics.Metric_Sum{
@@ -185,8 +213,9 @@ func intExemplarToExemplar(src []otlpmetrics.IntExemplar) []otlpmetrics.Exemplar
 	exemplars := []otlpmetrics.Exemplar{}
 	for _, exemplar := range src {
 		exemplars = append(exemplars, otlpmetrics.Exemplar{
-			FilteredLabels: exemplar.FilteredLabels,
-			TimeUnixNano:   exemplar.TimeUnixNano,
+			FilteredLabels:     exemplar.FilteredLabels,
+			FilteredAttributes: labelsToAttributes(exemplar.FilteredLabels),
+			TimeUnixNano:       exemplar.TimeUnixNano,
 			Value: &otlpmetrics.Exemplar_AsInt{
 				AsInt: exemplar.Value,
 			},
@@ -195,4 +224,37 @@ func intExemplarToExemplar(src []otlpmetrics.IntExemplar) []otlpmetrics.Exemplar
 		})
 	}
 	return exemplars
+}
+
+func numberDataPointsLabelsToAttributes(dps []*otlpmetrics.NumberDataPoint) {
+	for i := range dps {
+		if dps[i].Labels != nil && dps[i].Attributes != nil {
+			continue
+		}
+		if dps[i].Labels != nil {
+			dps[i].Attributes = labelsToAttributes(dps[i].Labels)
+		}
+	}
+}
+
+func summaryDataPointsLabelsToAttributes(dps []*otlpmetrics.SummaryDataPoint) {
+	for i := range dps {
+		if dps[i].Labels != nil && dps[i].Attributes != nil {
+			continue
+		}
+		if dps[i].Labels != nil {
+			dps[i].Attributes = labelsToAttributes(dps[i].Labels)
+		}
+	}
+}
+
+func histogramDataPointsLabelsToAttributes(dps []*otlpmetrics.HistogramDataPoint) {
+	for i := range dps {
+		if dps[i].Labels != nil && dps[i].Attributes != nil {
+			continue
+		}
+		if dps[i].Labels != nil {
+			dps[i].Attributes = labelsToAttributes(dps[i].Labels)
+		}
+	}
 }

--- a/model/internal/otlp_wrappers_test.go
+++ b/model/internal/otlp_wrappers_test.go
@@ -162,6 +162,9 @@ func TestDeprecatedIntHistogram(t *testing.T) {
 								TimeUnixNano:      1,
 								Count:             29,
 								Exemplars:         []otlpmetrics.Exemplar{},
+								Attributes: []otlpcommon.KeyValue{
+									{Key: "key", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "value"}}},
+								},
 							},
 						},
 					},
@@ -206,6 +209,9 @@ func TestDeprecatedIntHistogram(t *testing.T) {
 								TimeUnixNano:      3,
 								Count:             26,
 								Exemplars:         []otlpmetrics.Exemplar{},
+								Attributes: []otlpcommon.KeyValue{
+									{Key: "key2", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "value2"}}},
+								},
 							},
 						},
 					},
@@ -250,6 +256,9 @@ func TestDeprecatedIntHistogram(t *testing.T) {
 								TimeUnixNano:      3,
 								Count:             26,
 								Exemplars:         []otlpmetrics.Exemplar{},
+								Attributes: []otlpcommon.KeyValue{
+									{Key: "key2", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "value2"}}},
+								},
 							},
 						},
 					},
@@ -311,6 +320,9 @@ func TestDeprecatedIntGauge(t *testing.T) {
 								TimeUnixNano:      11,
 								Value:             &otlpmetrics.NumberDataPoint_AsInt{AsInt: 100},
 								Exemplars:         []otlpmetrics.Exemplar{},
+								Attributes: []otlpcommon.KeyValue{
+									{Key: "GaugeKey", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "GaugeValue"}}},
+								},
 							},
 						},
 					},
@@ -347,6 +359,9 @@ func TestDeprecatedIntGauge(t *testing.T) {
 								TimeUnixNano:      13,
 								Value:             &otlpmetrics.NumberDataPoint_AsInt{AsInt: 101},
 								Exemplars:         []otlpmetrics.Exemplar{},
+								Attributes: []otlpcommon.KeyValue{
+									{Key: "IntGaugeKey", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "IntGaugeValue"}}},
+								},
 							},
 						},
 					},
@@ -412,6 +427,9 @@ func TestDeprecatedIntSum(t *testing.T) {
 								TimeUnixNano:      21,
 								Value:             &otlpmetrics.NumberDataPoint_AsInt{AsInt: 200},
 								Exemplars:         []otlpmetrics.Exemplar{},
+								Attributes: []otlpcommon.KeyValue{
+									{Key: "SumKey", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "SumValue"}}},
+								},
 							},
 						},
 					},
@@ -452,6 +470,9 @@ func TestDeprecatedIntSum(t *testing.T) {
 								TimeUnixNano:      23,
 								Value:             &otlpmetrics.NumberDataPoint_AsInt{AsInt: 201},
 								Exemplars:         []otlpmetrics.Exemplar{},
+								Attributes: []otlpcommon.KeyValue{
+									{Key: "IntSumKey", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "IntSumValue"}}},
+								},
 							},
 						},
 					},
@@ -492,6 +513,9 @@ func TestDeprecatedIntSum(t *testing.T) {
 								TimeUnixNano:      23,
 								Value:             &otlpmetrics.NumberDataPoint_AsInt{AsInt: 201},
 								Exemplars:         []otlpmetrics.Exemplar{},
+								Attributes: []otlpcommon.KeyValue{
+									{Key: "IntSumKey", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "IntSumValue"}}},
+								},
 							},
 						},
 					},
@@ -528,11 +552,153 @@ func TestDeprecatedIntSum(t *testing.T) {
 								TimeUnixNano:      23,
 								Value:             &otlpmetrics.NumberDataPoint_AsInt{AsInt: 201},
 								Exemplars:         []otlpmetrics.Exemplar{},
+								Attributes: []otlpcommon.KeyValue{
+									{Key: "IntSumKey", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "IntSumValue"}}},
+								},
 							},
 						},
 					},
 				},
 			}},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.inputMetrics[0].Description, func(t *testing.T) {
+			req := &otlpcollectormetrics.ExportMetricsServiceRequest{
+				ResourceMetrics: []*otlpmetrics.ResourceMetrics{
+					{
+						InstrumentationLibraryMetrics: []*otlpmetrics.InstrumentationLibraryMetrics{
+							{
+								Metrics: test.inputMetrics,
+							},
+						}},
+				},
+			}
+			MetricsCompatibilityChanges(req)
+			assert.EqualValues(t, test.outputMetrics, req.ResourceMetrics[0].InstrumentationLibraryMetrics[0].Metrics)
+		})
+	}
+}
+
+func TestAttributesAndLabels(t *testing.T) {
+	tests := []struct {
+		inputMetrics  []*otlpmetrics.Metric
+		outputMetrics []*otlpmetrics.Metric
+	}{
+		{
+			inputMetrics: []*otlpmetrics.Metric{
+				{
+					// only Labels are set
+					Data: &otlpmetrics.Metric_Histogram{
+						Histogram: &otlpmetrics.Histogram{
+							AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA,
+							DataPoints: []*otlpmetrics.HistogramDataPoint{
+								{
+									Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+										{Key: "key", Value: "value"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			outputMetrics: []*otlpmetrics.Metric{
+				{
+					Data: &otlpmetrics.Metric_Histogram{
+						Histogram: &otlpmetrics.Histogram{
+							AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA,
+							DataPoints: []*otlpmetrics.HistogramDataPoint{
+								{
+									Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+										{Key: "key", Value: "value"},
+									},
+									Attributes: []otlpcommon.KeyValue{
+										{Key: "key", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "value"}}},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			inputMetrics: []*otlpmetrics.Metric{
+				{
+					// only Attributes are set
+					Data: &otlpmetrics.Metric_Histogram{
+						Histogram: &otlpmetrics.Histogram{
+							AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA,
+							DataPoints: []*otlpmetrics.HistogramDataPoint{
+								{
+									Attributes: []otlpcommon.KeyValue{
+										{Key: "key2", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "value2"}}},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			outputMetrics: []*otlpmetrics.Metric{
+				{
+					Data: &otlpmetrics.Metric_Histogram{
+						Histogram: &otlpmetrics.Histogram{
+							AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA,
+							DataPoints: []*otlpmetrics.HistogramDataPoint{
+								{
+									Attributes: []otlpcommon.KeyValue{
+										{Key: "key2", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "value2"}}},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			inputMetrics: []*otlpmetrics.Metric{
+				{
+					// both Attributes and Labels are set
+					Data: &otlpmetrics.Metric_Histogram{
+						Histogram: &otlpmetrics.Histogram{
+							AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA,
+							DataPoints: []*otlpmetrics.HistogramDataPoint{
+								{
+									Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+										{Key: "key1", Value: "value1"},
+									},
+									Attributes: []otlpcommon.KeyValue{
+										{Key: "key2", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "value2"}}},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			outputMetrics: []*otlpmetrics.Metric{
+				{
+					Data: &otlpmetrics.Metric_Histogram{
+						Histogram: &otlpmetrics.Histogram{
+							AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA,
+							DataPoints: []*otlpmetrics.HistogramDataPoint{
+								{
+									Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+										{Key: "key1", Value: "value1"},
+									},
+									Attributes: []otlpcommon.KeyValue{
+										{Key: "key2", Value: otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: "value2"}}},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
 		},
 	}
 

--- a/model/pdata/metrics_test.go
+++ b/model/pdata/metrics_test.go
@@ -268,12 +268,12 @@ func TestOtlpToInternalReadOnly(t *testing.T) {
 	assert.EqualValues(t, startTime, gaugeDataPoints.At(0).StartTimestamp())
 	assert.EqualValues(t, endTime, gaugeDataPoints.At(0).Timestamp())
 	assert.EqualValues(t, 123.1, gaugeDataPoints.At(0).DoubleVal())
-	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key0": "value0"}), gaugeDataPoints.At(0).LabelsMap())
+	assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{"key0": NewAttributeValueString("value0")}), gaugeDataPoints.At(0).Attributes())
 	// Second point
 	assert.EqualValues(t, startTime, gaugeDataPoints.At(1).StartTimestamp())
 	assert.EqualValues(t, endTime, gaugeDataPoints.At(1).Timestamp())
 	assert.EqualValues(t, 456.1, gaugeDataPoints.At(1).DoubleVal())
-	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key1": "value1"}), gaugeDataPoints.At(1).LabelsMap())
+	assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{"key1": NewAttributeValueString("value1")}), gaugeDataPoints.At(1).Attributes())
 
 	// Check double metric
 	metricDouble := metrics.At(1)
@@ -289,12 +289,12 @@ func TestOtlpToInternalReadOnly(t *testing.T) {
 	assert.EqualValues(t, startTime, sumDataPoints.At(0).StartTimestamp())
 	assert.EqualValues(t, endTime, sumDataPoints.At(0).Timestamp())
 	assert.EqualValues(t, 123.1, sumDataPoints.At(0).DoubleVal())
-	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key0": "value0"}), sumDataPoints.At(0).LabelsMap())
+	assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{"key0": NewAttributeValueString("value0")}), sumDataPoints.At(0).Attributes())
 	// Second point
 	assert.EqualValues(t, startTime, sumDataPoints.At(1).StartTimestamp())
 	assert.EqualValues(t, endTime, sumDataPoints.At(1).Timestamp())
 	assert.EqualValues(t, 456.1, sumDataPoints.At(1).DoubleVal())
-	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key1": "value1"}), sumDataPoints.At(1).LabelsMap())
+	assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{"key1": NewAttributeValueString("value1")}), sumDataPoints.At(1).Attributes())
 
 	// Check histogram metric
 	metricHistogram := metrics.At(2)
@@ -310,13 +310,13 @@ func TestOtlpToInternalReadOnly(t *testing.T) {
 	assert.EqualValues(t, startTime, histogramDataPoints.At(0).StartTimestamp())
 	assert.EqualValues(t, endTime, histogramDataPoints.At(0).Timestamp())
 	assert.EqualValues(t, []float64{1, 2}, histogramDataPoints.At(0).ExplicitBounds())
-	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key0": "value0"}), histogramDataPoints.At(0).LabelsMap())
+	assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{"key0": NewAttributeValueString("value0")}), histogramDataPoints.At(0).Attributes())
 	assert.EqualValues(t, []uint64{10, 15, 1}, histogramDataPoints.At(0).BucketCounts())
 	// Second point
 	assert.EqualValues(t, startTime, histogramDataPoints.At(1).StartTimestamp())
 	assert.EqualValues(t, endTime, histogramDataPoints.At(1).Timestamp())
 	assert.EqualValues(t, []float64{1}, histogramDataPoints.At(1).ExplicitBounds())
-	assert.EqualValues(t, NewStringMap().InitFromMap(map[string]string{"key1": "value1"}), histogramDataPoints.At(1).LabelsMap())
+	assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{"key1": NewAttributeValueString("value1")}), histogramDataPoints.At(1).Attributes())
 	assert.EqualValues(t, []uint64{10, 1}, histogramDataPoints.At(1).BucketCounts())
 }
 
@@ -351,7 +351,7 @@ func TestOtlpToFromInternalReadOnly(t *testing.T) {
 }
 
 func TestOtlpToFromInternalGaugeMutating(t *testing.T) {
-	newLabels := NewStringMap().InitFromMap(map[string]string{"k": "v"})
+	newAttributes := NewAttributeMap().InitFromMap(map[string]AttributeValue{"k": NewAttributeValueString("v")})
 
 	md := MetricsFromInternalRep(internal.MetricsFromOtlp(&otlpcollectormetrics.ExportMetricsServiceRequest{
 		ResourceMetrics: []*otlpmetrics.ResourceMetrics{
@@ -388,9 +388,9 @@ func TestOtlpToFromInternalGaugeMutating(t *testing.T) {
 	assert.EqualValues(t, endTime+1, gaugeDataPoints.At(0).Timestamp())
 	gaugeDataPoints.At(0).SetDoubleVal(124.1)
 	assert.EqualValues(t, 124.1, gaugeDataPoints.At(0).DoubleVal())
-	gaugeDataPoints.At(0).LabelsMap().Delete("key0")
-	gaugeDataPoints.At(0).LabelsMap().Upsert("k", "v")
-	assert.EqualValues(t, newLabels, gaugeDataPoints.At(0).LabelsMap())
+	gaugeDataPoints.At(0).Attributes().Delete("key0")
+	gaugeDataPoints.At(0).Attributes().UpsertString("k", "v")
+	assert.EqualValues(t, newAttributes, gaugeDataPoints.At(0).Attributes())
 
 	// Test that everything is updated.
 	assert.EqualValues(t, &otlpcollectormetrics.ExportMetricsServiceRequest{
@@ -409,10 +409,14 @@ func TestOtlpToFromInternalGaugeMutating(t *testing.T) {
 									Gauge: &otlpmetrics.Gauge{
 										DataPoints: []*otlpmetrics.NumberDataPoint{
 											{
-												Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+												Attributes: []otlpcommon.KeyValue{
 													{
-														Key:   "k",
-														Value: "v",
+														Key: "k",
+														Value: otlpcommon.AnyValue{
+															Value: &otlpcommon.AnyValue_StringValue{
+																StringValue: "v",
+															},
+														},
 													},
 												},
 												StartTimeUnixNano: startTime + 1,
@@ -434,7 +438,7 @@ func TestOtlpToFromInternalGaugeMutating(t *testing.T) {
 }
 
 func TestOtlpToFromInternalSumMutating(t *testing.T) {
-	newLabels := NewStringMap().InitFromMap(map[string]string{"k": "v"})
+	newAttributes := NewAttributeMap().InitFromMap(map[string]AttributeValue{"k": NewAttributeValueString("v")})
 
 	md := MetricsFromInternalRep(internal.MetricsFromOtlp(&otlpcollectormetrics.ExportMetricsServiceRequest{
 		ResourceMetrics: []*otlpmetrics.ResourceMetrics{
@@ -472,9 +476,9 @@ func TestOtlpToFromInternalSumMutating(t *testing.T) {
 	assert.EqualValues(t, endTime+1, doubleDataPoints.At(0).Timestamp())
 	doubleDataPoints.At(0).SetDoubleVal(124.1)
 	assert.EqualValues(t, 124.1, doubleDataPoints.At(0).DoubleVal())
-	doubleDataPoints.At(0).LabelsMap().Delete("key0")
-	doubleDataPoints.At(0).LabelsMap().Upsert("k", "v")
-	assert.EqualValues(t, newLabels, doubleDataPoints.At(0).LabelsMap())
+	doubleDataPoints.At(0).Attributes().Delete("key0")
+	doubleDataPoints.At(0).Attributes().UpsertString("k", "v")
+	assert.EqualValues(t, newAttributes, doubleDataPoints.At(0).Attributes())
 
 	// Test that everything is updated.
 	assert.EqualValues(t, &otlpcollectormetrics.ExportMetricsServiceRequest{
@@ -494,10 +498,14 @@ func TestOtlpToFromInternalSumMutating(t *testing.T) {
 										AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
 										DataPoints: []*otlpmetrics.NumberDataPoint{
 											{
-												Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+												Attributes: []otlpcommon.KeyValue{
 													{
-														Key:   "k",
-														Value: "v",
+														Key: "k",
+														Value: otlpcommon.AnyValue{
+															Value: &otlpcommon.AnyValue_StringValue{
+																StringValue: "v",
+															},
+														},
 													},
 												},
 												StartTimeUnixNano: startTime + 1,
@@ -519,7 +527,7 @@ func TestOtlpToFromInternalSumMutating(t *testing.T) {
 }
 
 func TestOtlpToFromInternalHistogramMutating(t *testing.T) {
-	newLabels := NewStringMap().InitFromMap(map[string]string{"k": "v"})
+	newAttributes := NewAttributeMap().InitFromMap(map[string]AttributeValue{"k": NewAttributeValueString("v")})
 
 	md := MetricsFromInternalRep(internal.MetricsFromOtlp(&otlpcollectormetrics.ExportMetricsServiceRequest{
 		ResourceMetrics: []*otlpmetrics.ResourceMetrics{
@@ -555,9 +563,9 @@ func TestOtlpToFromInternalHistogramMutating(t *testing.T) {
 	assert.EqualValues(t, startTime+1, histogramDataPoints.At(0).StartTimestamp())
 	histogramDataPoints.At(0).SetTimestamp(Timestamp(endTime + 1))
 	assert.EqualValues(t, endTime+1, histogramDataPoints.At(0).Timestamp())
-	histogramDataPoints.At(0).LabelsMap().Delete("key0")
-	histogramDataPoints.At(0).LabelsMap().Upsert("k", "v")
-	assert.EqualValues(t, newLabels, histogramDataPoints.At(0).LabelsMap())
+	histogramDataPoints.At(0).Attributes().Delete("key0")
+	histogramDataPoints.At(0).Attributes().UpsertString("k", "v")
+	assert.EqualValues(t, newAttributes, histogramDataPoints.At(0).Attributes())
 	histogramDataPoints.At(0).SetExplicitBounds([]float64{1})
 	assert.EqualValues(t, []float64{1}, histogramDataPoints.At(0).ExplicitBounds())
 	histogramDataPoints.At(0).SetBucketCounts([]uint64{21, 32})
@@ -579,10 +587,14 @@ func TestOtlpToFromInternalHistogramMutating(t *testing.T) {
 										AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA,
 										DataPoints: []*otlpmetrics.HistogramDataPoint{
 											{
-												Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+												Attributes: []otlpcommon.KeyValue{
 													{
-														Key:   "k",
-														Value: "v",
+														Key: "k",
+														Value: otlpcommon.AnyValue{
+															Value: &otlpcommon.AnyValue_StringValue{
+																StringValue: "v",
+															},
+														},
 													},
 												},
 												StartTimeUnixNano: startTime + 1,
@@ -750,10 +762,14 @@ func generateTestProtoGaugeMetric() *otlpmetrics.Metric {
 			Gauge: &otlpmetrics.Gauge{
 				DataPoints: []*otlpmetrics.NumberDataPoint{
 					{
-						Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+						Attributes: []otlpcommon.KeyValue{
 							{
-								Key:   "key0",
-								Value: "value0",
+								Key: "key0",
+								Value: otlpcommon.AnyValue{
+									Value: &otlpcommon.AnyValue_StringValue{
+										StringValue: "value0",
+									},
+								},
 							},
 						},
 						StartTimeUnixNano: startTime,
@@ -763,10 +779,14 @@ func generateTestProtoGaugeMetric() *otlpmetrics.Metric {
 						},
 					},
 					{
-						Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+						Attributes: []otlpcommon.KeyValue{
 							{
-								Key:   "key1",
-								Value: "value1",
+								Key: "key1",
+								Value: otlpcommon.AnyValue{
+									Value: &otlpcommon.AnyValue_StringValue{
+										StringValue: "value1",
+									},
+								},
 							},
 						},
 						StartTimeUnixNano: startTime,
@@ -790,10 +810,14 @@ func generateTestProtoSumMetric() *otlpmetrics.Metric {
 				AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
 				DataPoints: []*otlpmetrics.NumberDataPoint{
 					{
-						Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+						Attributes: []otlpcommon.KeyValue{
 							{
-								Key:   "key0",
-								Value: "value0",
+								Key: "key0",
+								Value: otlpcommon.AnyValue{
+									Value: &otlpcommon.AnyValue_StringValue{
+										StringValue: "value0",
+									},
+								},
 							},
 						},
 						StartTimeUnixNano: startTime,
@@ -803,10 +827,14 @@ func generateTestProtoSumMetric() *otlpmetrics.Metric {
 						},
 					},
 					{
-						Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+						Attributes: []otlpcommon.KeyValue{
 							{
-								Key:   "key1",
-								Value: "value1",
+								Key: "key1",
+								Value: otlpcommon.AnyValue{
+									Value: &otlpcommon.AnyValue_StringValue{
+										StringValue: "value1",
+									},
+								},
 							},
 						},
 						StartTimeUnixNano: startTime,
@@ -831,10 +859,14 @@ func generateTestProtoDoubleHistogramMetric() *otlpmetrics.Metric {
 				AggregationTemporality: otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA,
 				DataPoints: []*otlpmetrics.HistogramDataPoint{
 					{
-						Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+						Attributes: []otlpcommon.KeyValue{
 							{
-								Key:   "key0",
-								Value: "value0",
+								Key: "key0",
+								Value: otlpcommon.AnyValue{
+									Value: &otlpcommon.AnyValue_StringValue{
+										StringValue: "value0",
+									},
+								},
 							},
 						},
 						StartTimeUnixNano: startTime,
@@ -843,10 +875,14 @@ func generateTestProtoDoubleHistogramMetric() *otlpmetrics.Metric {
 						ExplicitBounds:    []float64{1, 2},
 					},
 					{
-						Labels: []otlpcommon.StringKeyValue{ //nolint:staticcheck // SA1019 ignore this!
+						Attributes: []otlpcommon.KeyValue{
 							{
-								Key:   "key1",
-								Value: "value1",
+								Key: "key1",
+								Value: otlpcommon.AnyValue{
+									Value: &otlpcommon.AnyValue_StringValue{
+										StringValue: "value1",
+									},
+								},
 							},
 						},
 						StartTimeUnixNano: startTime,

